app/main.py

```diff
@@ -0,0 +1,140 @@
+from typing import Optional
+from fastapi import FastAPI,Response,status,HTTPException
+from fastapi.params import Body
+from pydantic import BaseModel
+from random import randrange
+app=FastAPI()
+
+
+# class Post extends BaseModel
+
+
+class Post(BaseModel):
+    title: str
+    content: str
+    publisshed: bool = True 
+    rating: Optional[int]= None 
+
+
+# full optional field that use int value, but if user type str vaule field will be empty 
+# when we use the path parameter id will always be a string and that's why we have to change it to int
+# we have to ensure that it must be an integer and automatically converts it, if some early id that is not a string is entered, Greek is output
+# default value (optional field) if the user does not enter data, this will be the value of that field
+# complete optional field that receives an int but if the user does not enter it then it will be none (empty)
+
```

Comment:
I think these indents and new lines between comment section and function is not needed. I will merge this PR , but let's check code comments conventions.